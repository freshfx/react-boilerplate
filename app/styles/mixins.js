/* eslint-disable no-magic-numbers, capitalized-comments */
/* stylelint-disable property-no-vendor-prefix */
import {css} from '@emotion/core'
import {
  media,
  containerWidths,
  LINK_HOVER_COLOR,
  LINK_HIGHLIGHT_HOVER_COLOR,
  LINK_HIGHLIGHT_COLOR,
  LINK_COLOR,
  LINK_DECORATION,
  LINK_HOVER_DECORATION,
  SEARCHBAR_NAVBAR_MAX_WIDTH,
  WHITE,
  BLUE
} from './variables'

export const container = css`
  margin-left: auto;
  margin-right: auto;
  width: 100%;
  ${media.phone`width: ${containerWidths.phone}px;`}
  ${media.tablet`width: ${containerWidths.tablet}px;`}
  ${media.desktop`width: ${containerWidths.desktop}px;`}
  ${media.largeDesktop`width: ${containerWidths.largeDesktop}px;`}
`

export const cmsContainer = css`
  margin-left: auto;
  margin-right: auto;
  width: ${containerWidths.desktop - 250}px;
  ${media.largeDesktop`width: ${containerWidths.largeDesktop - 250}px;`}
`

export const smallerContainer = css`
  ${container}
  max-width: ${SEARCHBAR_NAVBAR_MAX_WIDTH};
`

const getColor = ({highlight = false}) => {
  if (highlight) {
    return LINK_HIGHLIGHT_COLOR
  }
  return LINK_COLOR
}

const getHoverColor = ({highlight = false}) => {
  if (highlight) {
    return LINK_HIGHLIGHT_HOVER_COLOR
  }
  return LINK_HOVER_COLOR
}

export const link = css`
  color: ${getColor};
  text-decoration: ${LINK_DECORATION};

  &:hover {
    color: ${getHoverColor};
    text-decoration: ${LINK_HOVER_DECORATION};
  }
`

export const ellipsis = css`
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
`

export const input = css`
  position: relative;
  width: 100%;
  font-size: 1rem;
  line-height: 1.5rem;
  color: ${WHITE};
  background-color: transparent;
  border: none;
  padding: 0;
  border-radius: 0;
  transition: color 0.4s ease, border-color 0.4s ease, background-color 0.4s ease;
  -webkit-appearance: none;
  -webkit-font-smoothing: antialiased;

  &:focus {
    outline: none;
    color: ${BLUE};
  }
`
