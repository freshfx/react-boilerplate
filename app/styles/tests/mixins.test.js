import React from 'react'
import styled from '@emotion/styled'
import {render} from '@testing-library/react'

import {link} from '../mixins'

const getComponent = mixin => styled.div`${mixin}`

describe('mixins', () => {
  describe('link', () => {
    const Component = getComponent(link)

    it('should render default style', () => {
      const {container} = render(<Component />)
      expect(container.firstChild).toMatchSnapshot()
    })

    it('should render highlight style', () => {
      const {container} = render(<Component highlight />)
      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
