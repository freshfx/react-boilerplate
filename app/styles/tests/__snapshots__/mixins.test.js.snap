// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mixins link should render default style 1`] = `
.emotion-0 {
  color: #222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:hover {
  color: #0A86CB;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
  class="emotion-0 emotion-1"
/>
`;

exports[`mixins link should render highlight style 1`] = `
.emotion-0 {
  color: #0A86CB;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:hover {
  color: #E40A43;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
  class="emotion-0 emotion-1"
/>
`;
